src/actions/collapseContext.ts

```diff
@@ -61,6 +62,14 @@ const collapseContext = (state: State, { at }: Options) => {
   const rankStart = getRankBefore(state, simplePath)
   const rankIncrement = (thought.rank - rankStart) / children.length
 
+  // Find attributes to delete
+  const pinAttributeId = findDescendant(state, head(simplePath), '=pin')
+  const childrenAttributeId = findDescendant(state, head(simplePath), '=children')
+  const childrenPinAttributeId = childrenAttributeId ? findDescendant(state, childrenAttributeId, '=pin') : null
+  const shouldDeleteChildrenAttribute =
+    childrenAttributeId &&
+    getChildren(state, childrenAttributeId).filter(thought => thought.value !== '=pin').length === 0
```

Comment:
This isn't significant here, but keep in mind short-circuiting when possible. You are iterating over the entire list when technically you can stop if `=pin` is found. [findAnyChild](https://github.com/cybersemics/em/blob/f545b318167c1f61b4af54c5d01fc35c22b231d9/src/selectors/getChildren.ts#L172) is a useful helper method for this.

Another approach would be to do the check *after* `=children/=pin` is deleted. That way you don't need to do the filter at all. You can just delete `=children` if it is empty at that point. (Though note that you would need to use the uncurried `deleteThought` to get access to the updated`state`.)

This is good enough as-is, just thought I'd point out some alternatives.